src/terminal_pane/terminal_pane.rs

```diff
@@ -471,15 +466,12 @@ impl vte::Perform for TerminalPane {
                         self.cursor_key_mode = false;
                     }
                     _ => {}
-                };
+                }
+            } else if params.get(0) == Some(&4) {
+                self.scroll.clear_all();
             }
         } else if c == 'h' {
-            let first_intermediate_is_questionmark = match _intermediates.get(0) {
-                Some(b'?') => true,
-                None => false,
-                _ => false,
-            };
-            if first_intermediate_is_questionmark {
+            if _intermediates.get(0) == Some(&b'?') {
```

Comment:
Also - thanks for this. This is of course much better. I should not be coding in my sleep ;P